ccmlib/scylla_node.py

```diff
@@ -729,19 +729,9 @@ def do_stop(self, gently=True):
         return did_stop
 
     def _wait_until_stopped(self, wait_seconds):
-        start_time = time.time()
-        wait_time_sec = 1
-        while True:
-            if not self.is_running():
-                return True
-            elapsed = time.time() - start_time
-            if elapsed >= wait_seconds:
-                return False
-            time.sleep(wait_time_sec)
-            if elapsed + wait_time_sec > wait_seconds:
-                wait_time_sec = wait_seconds - elapsed
-            elif wait_time_sec <= 16:
-                wait_time_sec *= 2
+        if wait_for(func=self.is_running, timeout=wait_seconds):
+            return True
+        return False
```

Comment:
If you fix wait_for() like I suggested above, you wouldn't need this function at all :-(
If you don't, instead of an if I think you can simply do
```python
return bool(wait_for(...))
```

Reply:
Done